var/spack/repos/builtin/packages/openpmd-api/package.py

```diff
@@ -105,10 +107,12 @@ def cmake_args(self):
                 self.define('openPMD_USE_INTERNAL_PYBIND11', False)
             ]
 
-        args += [
-            self.define('openPMD_USE_INTERNAL_JSON', False),
-            self.define('openPMD_USE_INTERNAL_VARIANT', False)
-        ]
+        args.append(self.define('openPMD_USE_INTERNAL_JSON', False))
+        if spec.satisfies('@:0.14.99'):  # pre C++17 releases
+            args.append(self.define('openPMD_USE_INTERNAL_VARIANT', False))
+        if spec.satisfies('@0.15.0:'):
```

Comment:
In case they add a beta version ;)
```suggestion
        if spec.satisfies('@0.15:'):
```

Reply:
Haha, I know me and I go straight to releases :D